### ISSUE TITLE
Add keepcart type functionality

### ISSUE THREAD

**Issue by lankeeyankee** [zencart/zencart#2576]
The ability to set a cookie to allow a guest's cart to be saved for admin-specified time is very handy. We have been using this since it's initial release and it has boosted sales since users can come back weeks later and still have an intact cart instead of having to start over again. It is pretty small and seems unobtrusive and I think it would be useful to add to the core.

**Comment by drbyte** [comment]
I seem to recall there being 2 different addons for this.
Opinions on which is most reliable, least intrusive, and most user friendly?

**Comment by scottcwilson** [comment]
KeepCart is cookie based.  It works for guests, but only on the same computer. 
Save For Later is database based.   It doesn't work for guests, but works for any computer. 

Keep Cart: https://www.zen-cart.com/downloads.php?do=file&id=992
Save for Later: https://www.zen-cart.com/downloads.php?do=file&id=872

Save for Later has had a lot more use, but it's a very complicated mod.

**Comment by lankeeyankee** [comment]
Yeah, keepcart is the one I am referring to. It's a cookie and I've been using it without any known issues for 10 years. It creates 3 extra settings in config>seesions, the admin can turn the cookie on/off, set how long the cart is good for in days, and a secret key as a security measure. Very small, an auto_loader and class/observer.

**Comment by drbyte** [comment]
Yes, putting KeepCart into core makes sense.

A couple suggestions:
- change the `change me` string value in the Install/Upgrade SQL to auto create a random string, using something like `SUBSTRING(MD5(RAND()) FROM 1 FOR 12)`
- Remove trailing `?>` from last line/s of files

**Comment by scottcwilson** [comment]
Related: #4273

**Comment by lat9** [comment]
> Yes, putting KeepCart into core makes sense.
> 
> A couple suggestions:
> 
>     * change the `change me` string value in the Install/Upgrade SQL to auto create a random string, using something like `SUBSTRING(MD5(RAND()) FROM 1 FOR 12)`
> 
>     * Remove trailing `?>` from last line/s of files

Changing the `change me` to something random is that the storefront processing (see https://github.com/lat9/keep_cart) checks to see that the value has (er) changed from the default value and disables the Keep Cart processing if it hasn't been changed.
